vllm/v1/sample/rejection_sampler.py

```diff
@@ -131,6 +141,123 @@ def parse_output(
         ]
         return outputs
 
+    def apply_logits_processors(
+        self,
+        logits: torch.Tensor,
+        sampling_metadata: SamplingMetadata,
+        metadata: SpecDecodeMetadata,
+    ):
+        any_penalties_or_bad_words = (sampling_metadata.bad_words_token_ids
+                                      or not sampling_metadata.no_penalties)
+
+        output_token_ids = sampling_metadata.output_token_ids
+        if any_penalties_or_bad_words:
+            output_token_ids = self._combine_outputs_with_spec_tokens(
+                sampling_metadata.output_token_ids,
+                sampling_metadata.spec_token_ids,
+            )
+        # Calculate indices of target logits
+        if sampling_metadata.allowed_token_ids_mask is not None \
+            or not sampling_metadata.no_penalties:
+            num_requests = len(sampling_metadata.output_token_ids)
+            num_draft_tokens = torch.tensor(metadata.num_draft_tokens,
+                                            device="cpu")
+            original_indices = torch.arange(num_requests, device="cpu")
+            repeat_indices_cpu = torch.repeat_interleave(
+                original_indices, repeats=num_draft_tokens)
+            repeat_indices = repeat_indices_cpu.to(device=logits.device,
+                                                   non_blocking=True)
+        else:
+            # Not use them for other cases
+            repeat_indices = None
+
+        logits = self.apply_penalties(logits, sampling_metadata, metadata,
+                                      repeat_indices, output_token_ids)
+        logits = self.apply_bad_words(logits, sampling_metadata, metadata,
+                                      output_token_ids)
+        logits = self.apply_allowed_token_ids(logits, sampling_metadata,
+                                              repeat_indices)
```

Comment:
Can simplify here I think
```suggestion
            logits = self.apply_penalties(logits, sampling_metadata, metadata,
                                  repeat_indices, output_token_ids)
            logits = self.apply_allowed_token_ids(logits, sampling_metadata,
                                                  repeat_indices)

        logits = self.apply_bad_words(logits, sampling_metadata, metadata,
                                      output_token_ids)
```